### PR TITLE
some docs for Sources

### PR DESCRIPTION
**Pull request by westnordost**
also,
- added the tileSize parameter for RasterSource
- remember*Source return typed *

### CODE REVIEW

**Comment by westnordost** [comment]
I just documented the status quo, you may want to check if you think the type of exceptions thrown is what you want.

File: lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt [hunk 1]
```diff
@@ -4,30 +4,40 @@ import androidx.compose.runtime.Composable
...
+ * @throws IllegalArgumentException if a layer with the given [id] already exists.
+ */
 @Composable
 @Suppress("NOTHING_TO_INLINE")
 public inline fun rememberGeoJsonSource(
   id: String,
   uri: String,
   options: GeoJsonOptions = GeoJsonOptions(),
-): Source =
+): GeoJsonSource =
```
Comment (sargunv) [inline lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt:21]: Okay to merge for now but I think we'll probably want to split these types into mutable and non mutable interfaces, and return the non mutable interface here so there's only one path for setting the uri (via the Composable parameter). 

Making mutations internal would also work, but I'd like to split the modules between core and compose so internal wouldn't work forever

Comment (westnordost) [reply lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt:21]: Why should Source(s) be mutable at all?

Comment (sargunv) [reply lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt:21]: The geojson data can be updated. We use that in the demo app right now on the camera follow demo, to move the location puck.

In compose that's done with a parameter to the Composable, but the underlying non-compose model just has a setter for it.

Comment (westnordost) [reply lib/maplibre-compose/src/commonMain/kotlin/dev/sargunv/maplibrecompose/compose/source/rememberGeoJsonSource.kt:21]: Oh, right! I didn't look at the code, I wrongly assumed that (in compose) the whole GeoJsonSource would be replaced on any change to it.

**Pull request merged**